src/template.js

```diff
@@ -35,6 +38,39 @@ function logger(message, event) {
   log('[GTM-DEBUG] Search Insights > ' + message, event || '');
 }
 
+function shallowObjectClone(obj) {
+  const keys = Object.keys(obj);
+  const newObj = {};
+  keys.forEach((key) => {
+    newObj[key] = obj[key];
+  });
+  return newObj;
+}
+
+function chunkPayload(payload, keys, limit) {
+  // check if the values in `payload` for each of `keys` have the same length.
+  const sameNumberOfValues = keys
+    .map((k) => payload[k].length)
+    .every((n) => n === payload[keys[0]].length);
```

Comment:
Actually, I do not know whether the `every` method is available in the context of GTM. Do you know if and how we could verify this?

Reply:
I tried it and it's accepted. The way I tested is, on my example app, I "forked" Algolia insights template. That way I can modify the code on the GTM dashboard. And as I updated the part, GTM didn't throw any error to me. It does when I use some unsupported syntax in their sandboxed JS environment.


![Screenshot 2021-12-08 at 14 59 43](https://user-images.githubusercontent.com/499898/145221148-72eae909-f8a9-46b3-9717-713c9c6c6b34.gif)

Reply:
Technically, it could be valid syntax even if the function does not exist right? because the prototype of array could be patched later on to add the `every` method, which the syntax checker would be able to detect without running the actual code, so it should let it run.

But looking at https://caniuse.com/?search=every, I think if it relies on browser behaviour for this method, we should be good to go.

Reply:
Yeah, I think they're relying on the browser behaviors in this case. 

For example, `typeof aaa` -> this doesn't work and they provide a dedicated function:
https://developers.google.com/tag-platform/tag-manager/templates/api#gettype